tests/integration/test_ha.py

```diff
@@ -2,6 +2,8 @@
 # Copyright 2024 Canonical Limited
 # See LICENSE file for licensing details.
 
+# TODO: Revisit this test after recent updates in the purpose of Kyuubi <> Zookeeper relation
```

Comment:
What do you mean here ?

Reply:
Before K8s exposure, HA used to be tied up with Zookeeper because we used Zookeeper for service discovery. This is no longer the case. We use Zookeeper only for engine discovery at the moment, and thus our tests involving zookeeper need to be updated for this particular role of zookeeper. For example, we don't test the "engine discovery" behavior of zookeeper.